sdk/servicebus/Azure.Messaging.ServiceBus/tests/Transactions/TransactionLiveTests.cs

```diff
@@ -958,6 +964,83 @@ public async Task CrossEntityTransactionSessionProcessor()
             _ = await noTxClient.AcceptNextSessionAsync(queueC.QueueName);
         }
 
+        [Test]
+        public async Task CrossEntityTransactionConnectionDropped()
+        {
+            await using var client = CreateCrossEntityTxnClient();
+            await using var queueA = await ServiceBusScope.CreateWithQueue(enablePartitioning: false, enableSession: false);
+            await using var queueB = await ServiceBusScope.CreateWithQueue(enablePartitioning: false, enableSession: false);
+
+            await using var noTxClient = CreateNoRetryClient();
+            var senderA = noTxClient.CreateSender(queueA.QueueName);
+            await senderA.SendMessageAsync(new ServiceBusMessage());
+
+            var receiverA = client.CreateReceiver(queueA.QueueName);
+            var senderB = client.CreateSender(queueB.QueueName);
+
+            var receivedMessage = await receiverA.ReceiveMessageAsync();
+            using (var ts = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))
+            {
+                await receiverA.CompleteMessageAsync(receivedMessage);
+
+                SimulateNetworkFailure(client, receiverA);
+                Assert.That(
+                    async () => await senderB.SendMessageAsync(new ServiceBusMessage()),
+                    Throws.InstanceOf<InvalidOperationException>());
+            }
+
+            // allow enough time for the service to discard the transaction
+            receivedMessage = await receiverA.ReceiveMessageAsync(TimeSpan.FromSeconds(200));
+            using (var ts = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))
+            {
+                await receiverA.CompleteMessageAsync(receivedMessage);
+                await senderB.SendMessageAsync(new ServiceBusMessage());
+                ts.Complete();
+            }
+
+            var receiverB = noTxClient.CreateReceiver(queueB.QueueName);
+            Assert.IsNotNull(await receiverB.ReceiveMessageAsync());
+        }
+
+        private static void SimulateNetworkFailure(ServiceBusClient client, ServiceBusReceiver receiver)
```

Comment:
Just know that I feel bad about this..

Reply:
You have my sympathies, sir.

I don't normally have much of an issue reflecting on our own types to avoid leaking a ton of private state as `internal` just for a couple of tests.   The reaching into the AMQP library types is regrettable, but the library is pretty much entirely unmockable... so this does avoid a ton of indirection by helper methods.

Reply:
You should.

Reply:
I feel less bad about now after https://github.com/Azure/azure-sdk-for-net/pull/21531/commits/b621e7046411e616d81cc736df9112d1e28c70ea

Thanks @pakrym for pointing out AbortInternal!